google/cloud/bigquery/schema.py

```diff
@@ -263,6 +279,15 @@ def from_api_repr(cls, api_repr: dict) -> "SchemaField":
         """
         placeholder = cls("this_will_be_replaced", "PLACEHOLDER")
 
+        # The API would return a string despite we send an integer. To ensure
+        # success of resending received schema, we convert string to integer
+        # to ensure consistency.
+        if (
+            isinstance(api_repr, dict)
+            and type(api_repr.get("timestampPrecision")) is str
+        ):
+            api_repr["timestampPrecision"] = int(api_repr["timestampPrecision"])
+
```

Comment:
suggestion: this might be cleaner with a try/catch:

```
try:
  # ensure timestamp is in int format
  api_repr["timestampPrecision"] = int(api_repr["timestampPrecision"])
except TypeError, KeyError:
  # ignore unset timestamps
  pass
```